src/Features/Core/Portable/InitializeParameter/AbstractInitializeParameterCodeRefactoringProvider.cs

```diff
@@ -28,62 +29,126 @@ internal abstract partial class AbstractInitializeParameterCodeRefactoringProvid
         protected abstract SyntaxNode GetBody(SyntaxNode functionDeclaration);
         protected abstract SyntaxNode GetTypeBlock(SyntaxNode node);
 
+        protected virtual ImmutableArray<SyntaxNode> GetParameters(SyntaxNode parameterNode)
+        {
+            return default;
+        }
         protected abstract void InsertStatement(
             SyntaxEditor editor, SyntaxNode functionDeclaration, IMethodSymbol method,
             SyntaxNode statementToAddAfterOpt, TStatementSyntax statement);
 
-        protected abstract Task<ImmutableArray<CodeAction>> GetRefactoringsAsync(
+        protected abstract Task<ImmutableArray<CodeAction>> GetRefactoringsForAllParametersAsync(
             Document document, IParameterSymbol parameter, SyntaxNode functionDeclaration, IMethodSymbol method,
             IBlockOperation blockStatementOpt, CancellationToken cancellationToken);
 
         public override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
         {
-            var (document, textSpan, cancellationToken) = context;
+            var position = context.Span.Start;
+            var document = context.Document;
+            var cancellationToken = context.CancellationToken;
             var syntaxTree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
             var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var token = root.FindToken(position);
+            var firstparameterNode = GetParameterNode(token, position);
 
-            var parameterNode = await context.TryGetRelevantNodeAsync<TParameterSyntax>().ConfigureAwait(false);
-            if (parameterNode == null)
+            var parameterNodes = GetParameters(firstparameterNode);
+            if (parameterNodes.IsDefault)
             {
                 return;
             }
 
-            var functionDeclaration = parameterNode.FirstAncestorOrSelf<SyntaxNode>(IsFunctionDeclaration);
-            if (functionDeclaration == null)
+            // List with parameterNodes that pass all checks
+            var listOfPotentiallyValidParametersNodes = new ImmutableArray<SyntaxNode>();
```

Comment:
Don't use the default constructor of `immutableArray`, it will create one with default value (and will crash when used). Use `Microsoft.CodeAnalysis.PooledObjects.ArrayBuilder` for constructing `ImmutableArray`.